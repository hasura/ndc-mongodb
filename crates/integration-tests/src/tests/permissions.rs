use crate::graphql_query;
use insta::assert_yaml_snapshot;

#[tokio::test]
async fn filters_results_according_to_configured_permissions() -> anyhow::Result<()> {
    assert_yaml_snapshot!(
        graphql_query(
            r#"
            query {
              users(order_by: {id: Asc}) {
                id
                name
                email
                comments(limit: 5, order_by: {id: Asc}) {
                  date
                  email
                  text
                }
              }
              comments(limit: 5, order_by: {id: Asc}) {
                date
                email
                text
              }
            }
            "#
        )
        .headers([
            ("x-hasura-role", "user"),
            ("x-hasura-user-id", "59b99db4cfa9a34dcd7885b6"),
        ])
        .run()
        .await?
    );
    Ok(())
}
