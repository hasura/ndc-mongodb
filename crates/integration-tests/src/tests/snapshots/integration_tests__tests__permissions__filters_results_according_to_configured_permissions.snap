---
source: crates/integration-tests/src/tests/permissions.rs
expression: "graphql_query(r#\"\n            query {\n              users(limit: 5) {\n                id\n                name\n                email\n                comments(limit: 5) {\n                  date\n                  email\n                  text\n                }\n              }\n              comments(limit: 5) {\n                date\n                email\n                text\n              }\n            }\n            \"#).headers([(\"x-hasura-role\",\n                            \"user\"),\n                        (\"x-hasura-user-id\",\n                            \"59b99db4cfa9a34dcd7885b6\")]).run().await?"
---
data:
  users:
    - id: 59b99db4cfa9a34dcd7885b6
      name: Ned Stark
      email: sean_bean@gameofthron.es
      comments:
        - date: "2000-01-21T03:17:04.000000000Z"
          email: sean_bean@gameofthron.es
          text: Illo nostrum enim sequi doloremque dolore saepe beatae. Iusto alias odit quaerat id dolores. Dolore quaerat accusantium esse voluptatibus. Aspernatur fuga exercitationem explicabo.
        - date: "2005-09-24T16:22:38.000000000Z"
          email: sean_bean@gameofthron.es
          text: Architecto eos eum iste facilis. Sunt aperiam fugit nihil quas.
        - date: "1978-10-22T23:49:33.000000000Z"
          email: sean_bean@gameofthron.es
          text: Aspernatur ullam blanditiis qui dolorum. Magnam minima suscipit esse. Laudantium voluptates incidunt quia saepe.
        - date: "2013-08-15T07:24:54.000000000Z"
          email: sean_bean@gameofthron.es
          text: Ullam error officiis incidunt praesentium debitis. Rerum repudiandae illum reprehenderit aut non. Iusto eum autem veniam eveniet temporibus sed. Accusamus sint sed veritatis eaque.
        - date: "2004-12-22T12:53:43.000000000Z"
          email: sean_bean@gameofthron.es
          text: Ducimus sunt neque sint nesciunt quis vero. Debitis ex non asperiores voluptatem iusto possimus. Doloremque blanditiis consequuntur explicabo placeat commodi repudiandae.
  comments:
    - date: "2000-01-21T03:17:04.000000000Z"
      email: sean_bean@gameofthron.es
      text: Illo nostrum enim sequi doloremque dolore saepe beatae. Iusto alias odit quaerat id dolores. Dolore quaerat accusantium esse voluptatibus. Aspernatur fuga exercitationem explicabo.
    - date: "2005-09-24T16:22:38.000000000Z"
      email: sean_bean@gameofthron.es
      text: Architecto eos eum iste facilis. Sunt aperiam fugit nihil quas.
    - date: "1978-10-22T23:49:33.000000000Z"
      email: sean_bean@gameofthron.es
      text: Aspernatur ullam blanditiis qui dolorum. Magnam minima suscipit esse. Laudantium voluptates incidunt quia saepe.
    - date: "2013-08-15T07:24:54.000000000Z"
      email: sean_bean@gameofthron.es
      text: Ullam error officiis incidunt praesentium debitis. Rerum repudiandae illum reprehenderit aut non. Iusto eum autem veniam eveniet temporibus sed. Accusamus sint sed veritatis eaque.
    - date: "2004-12-22T12:53:43.000000000Z"
      email: sean_bean@gameofthron.es
      text: Ducimus sunt neque sint nesciunt quis vero. Debitis ex non asperiores voluptatem iusto possimus. Doloremque blanditiis consequuntur explicabo placeat commodi repudiandae.
errors: ~
